libs/wingc/src/jsify.rs

```diff
@@ -1490,13 +1557,25 @@ impl Display for Component<'_> {
 }
 
 impl<'a> FieldReferenceVisitor<'a> {
-	fn analyze_expr(node: &'a Expr) -> Vec<Component> {
+	fn analyze_expr(&self, node: &'a Expr) -> Vec<Component> {
 		match &node.kind {
 			ExprKind::Reference(Reference::Identifier(x)) => {
+				let scope = match &self.function_def.body {
```

Comment:
if this is a free variable, we still end up analyzing the expression. I wasn't sure what to rename this piece of our compiler to instead of `FieldReferenceVisitor`. My current idea is that this visitor looks inside functions for expressions that require bindings to be generated, and collects the operations that are used on them in `references` (or generates diagnostics if the expression shouldn't be possible to capture) - it's no longer limiting the analysis to fields anymore.